envoy/network/connection.h

```diff
@@ -205,6 +221,14 @@ class Connection : public Event::DeferredDeletable,
    */
   virtual void detectEarlyCloseWhenReadDisabled(bool should_detect) PURE;
 
+  /**
+   * Set if Envoy should detect TCP Rst
+   *
+   * @param enbale supplies if disconnects should be detected when the connection has been
+   * read disabled
+   */
+  virtual void enableTcpRstDetectAndSend(bool enbale) PURE;
```

Comment:
Is this controlled by the runtime option only? Or do we have some other uses for this API. If this is set only from the runtime value, I suggest removing the API and just read the runtime flag in the connection object.

Reply:
No, I think we may have other uses for this API if they don't want to detect RST, like `enableHalfClose` api. Under this API it is also controlled by the runtime guard. It maybe redundant.

Reply:
I think it should be ok to always detect RST. If something up the chain is interested in it it would call the `detectedCloseType()`. Otherwise it is just there. There is no significant cost in detecting the close type.

Reply:
Done, thanks!